### ISSUE TITLE
Rocket Trajectory Optimization is missing the last two graphs

### ISSUE THREAD

**Issue by dvd101x** [josdejong/mathjs#2873]
While reading the source code of **rocket trajectory optimization** I noticed that there are two graphs that are not shown on the rendered webpage.

- Drag Acceleration
- Earth Chart

The issue seems to start at line 194
[examples/browser/rocket_trajectory_optimization.html](https://github.com/josdejong/mathjs/blob/f646b7eb100ac8553b665e5daf25b45524aada13/examples/browser/rocket_trajectory_optimization.html#L194)

To reproduce just open the example on the browser and check the console log
https://mathjs.org/examples/browser/rocket_trajectory_optimization.html

I have a proposal ready to fix this issue and update to the latest **chartjs** and some simplifications.

Maybe I changed too much stuff so I'm not sure if I should make a pull request
https://github.com/dvd101x/mathjs/blob/f9735c999aae5ff9739657f7e89d69ead59cf290/examples/browser/rocket_trajectory_optimization.html

I also worked on fix that keeps the same chartjs version and has less changes
https://gist.github.com/dvd101x/61d4ca84da4a9cf6d026155eb7280e8a#file-rocket_trajectory_optimization_update-html

**Comment by josdejong** [comment]
Thanks for bringing this up David.

So what is the cause exactly? What is a minimal fix? The error looks like the breaking change #2440 introduced in in v11.0.0, but from your adjusted example it's hard to see the exact change to fix the original example.

**Comment by dvd101x** [comment]
Hi Jos, thank you for looking into it!

You found exactly the issue

Unlike matlab or numpy, on mathjs when we try to use a vector in an exponential we get an error.
```
exp([1,2])

TypeError: Function 'exp' doesn't apply to matrices. To call it elementwise on a matrix 'M', try 'map(M, exp)'.
```

To calculate the *drag acceleration* it is expected to evaluate at some point the function drag with vectors `drag(result[:, 1], result[:, 2])`, the function drag `drag(r, v) = 1/2 .* density(r) .* v.^2 * A * dragCoef` depends on the function density `density(r) = 1.2250 kg/m^3 * exp(-g0 * (r - r0) / (83246.8 m^2/s^2))`. Density requires an exponential.

The minimum fix that I've found is to avoid sending vector to the function drag, so instead of having

``` javascript
sim.evaluate("concat(drag(result[:, 1], result[:, 2]) ./ result[:, 3], result[:, 6])")
        .toArray()
        .map(([dragAcc, t]) => ({ x: t.toNumber('s'), y: dragAcc.toNumber('m/s^2') }))
```
We can use

```javascript
sim.evaluate('result')
          .toArray()
          .map(([r, v, m, phi, gamma, t]) => ({
            x: t.toNumber('s'),
            y: sim.evaluate(`drag(${r},${v})/${m}`).toNumber('m/s^2')
          }))
```

Or going back to a version of mathjs that can solve `exp([1,2])`

**Comment by dvd101x** [comment]
Here is a simpler fix

Make a new function that can calculate exponentials for both scalars and arrays `Exp(A) =  equalText(typeOf(A), 'number') ? exp(A) : map(A, exp)`

And use it in the function for density `density(r) = 1.2250 kg/m^3 * Exp(-g0 * (r - r0) / (83246.8 m^2/s^2))`

Thus all the rest of the logic gets preserved.

**Comment by josdejong** [comment]
Fix is published now in `v11.5.1`.
